src/main/java/com/puppycrawl/tools/checkstyle/utils/TokenUtil.java

```diff
@@ -319,4 +320,35 @@ public static boolean isBooleanLiteralType(final int tokenType) {
         return isTrue || isFalse;
     }
 
+    /**
+     * Get name of package and super class of anon inner class by concatenating
+     * the identifier values under {@link TokenTypes#DOT}.
+     *
+     * @param ast ast to extract superclass or package name from
+     * @return qualified name
+     */
+    public static String extractQualifiedName(DetailAST ast) {
+        return FullIdent.createFullIdent(ast).getText();
+    }
+
+    /**
+     * Get the short name of super class of anonymous inner class.
+     * Example-
+     * <pre>
+     * TestClass.NestedClass obj = new Test().new NestedClass() {};
+     * // Short name will be Test.NestedClass
+     * </pre>
+     *
+     * @param literalNewAst ast node of type {@link TokenTypes#LITERAL_NEW}
+     * @return short name of base class of anonymous inner class
+     */
+    public static String getShortNameOfAnonInnerClass(DetailAST literalNewAst) {
```

Comment:
This is currently being used only by UnusedLocalVar but will be used by FinalClass for fixing #11365